apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -71,6 +71,22 @@ function cloneThread(thread) {
   };
 }
 
+function clone(objectToBeCloned) {
+  // Basis checking.
+  if (!(objectToBeCloned instanceof Object)) {
+    return objectToBeCloned;
+  }
+
+  var objectClone = {};
+
+  // Clone each property.
+  for (var prop in objectToBeCloned) {
```

Comment:
maybe with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames ? but I wonder if some properties come from the prototype ... or some properties that we don't want..